include/cuco/static_map.cuh

```diff
@@ -606,13 +613,37 @@ class static_map {
      * equality
      * @return `true` if the insert was successful, `false` otherwise.
      */
-    template <typename Hash     = cuco::detail::MurmurHash3_32<key_type>,
-              typename KeyEqual = thrust::equal_to<key_type>>
-    __device__ bool insert(value_type const& insert_pair,
-                           Hash hash          = Hash{},
-                           KeyEqual key_equal = KeyEqual{}) noexcept;
+    template <typename pair_type = value_type,
+              typename Hash      = cuco::detail::MurmurHash3_32<key_type>,
+              typename KeyEqual  = thrust::equal_to<key_type>>
+    __device__ std::enable_if_t<cuco::detail::is_packable<pair_type>(), bool> insert(
```

Comment:
Having the `enable_if` on the public interface is less than ideal. Perhaps we can move that to an internal `insert_impl` function template that is called by the public `insert`.